mapie/control_risk/ltt.py

```diff
@@ -0,0 +1,103 @@
+import numpy as np
+import warnings
+from numpy.typing import NDArray
+from typing import Tuple, List, Union, Optional, Any
+from .p_values import hoefdding_bentkus_p_value
+
+
+def _ltt_procedure(
+    r_hat: NDArray,
+    alpha_np: NDArray,
+    delta: Optional[float],
+    n_obs: int
+) -> Tuple[List[List[Any]], NDArray]:
+    """
+    Apply the ltt procedure for risk control
+    should be precision for multilabelclassif
+    Note that we will do a multipletest for r_hat that are
+    less than level alpha.
+    Procedure:
+        - compute p_values for each lambdas descretize
+        - Apply a fwer algorithm, here Bonferonni correction
+        - Return the index lambdas that give you the control
+        at alpha level
+        - n_obs is the length of data cal
+    Parameters
+    ----------
+    r_hat: NDArray of shape (n_samples, )
+        Empirical risk of metric_control with respect
+        to the lambdas.
+    alpha: NDArray of alphas level.
+    delta: Float value
+        Correspond to proportion of failure we don't
+        want to exceed.
+    Returns
+    ----------
+    valid_index: NDArray of shape (n_alpha, ).
+        Contain the valid index that satisfy fwer control
+        for each alpha (shape aren't the same for each alpha)
+    p_values: NDArray of shape (n_lambda, n_alpha)
+        Contains the values of p_value for different alpha
+    """
```

Comment:
STYLE: 
- add newlines between parameters and sub-sections (parameters, returns, ...)
- no abbreviation, complete words (l15 ltt -> learn then test, l16: multilabelclassif -> multi-label classification)
- remove l24
-  decompose de type and the description (l30)
ENH: add the paper reference of LTT.

Reply:
Done